Bio/Align/__init__.py

```diff
@@ -926,6 +928,429 @@ def sort(self, key=None, reverse=False):
+        else:
+            query = query.seq
+        try:
+            Tname = target.id
+        except AttributeError:
+            Tname = "target"
+        else:
+            target = target.seq
+        seq1 = str(target)
+        seq2 = str(query)
+        n1 = len(seq1)
+        n2 = len(seq2)
+        match = 0
+        mismatch = 0
+        repmatch = 0
+        Ns = 0
+        Qgapcount = 0
+        Qgapbases = 0
+        Tgapcount = 0
+        Tgapbases = 0
+        Qsize = n2
+        Qstart = 0
+        Qend = Qsize
+        Tsize = n1
+        Tstart = 0
+        Tend = Tsize
+        blockSizes = []
+        qStarts = []
+        tStarts = []
+        strand = '+'
+        start1 = 0
+        start2 = 0
+        start1, start2 = self.path[0]
+        for end1, end2 in self.path[1:]:
+            count1 = end1 - start1
+            count2 = end2 - start2
+            if count1 == 0:
+                if start2 == 0:
+                    Qstart += count2
+                elif end2 == n2:
+                    Qend -= count2
+                else:
+                    Qgapcount += 1
+                    Qgapbases += count2
+                start2 = end2
+            elif count2 == 0:
+                if start1 == 0:
+                    Tstart += count1
+                elif end1 == n1:
+                    Tend -= count1
+                else:
+                    Tgapcount += 1
+                    Tgapbases += count1
+                start1 = end1
+            else:
+                assert count1 == count2
+                tStarts.append(start1)
+                qStarts.append(start2)
+                blockSizes.append(count1)
+                for c1, c2 in zip(seq1[start1:end1], seq2[start2:end2]):
+                    if c1 == 'N' or c2 == 'N':
+                        Ns += 1
+                    elif c1 == c2:
+                        match += 1
+                    else:
+                        mismatch += 1
+                start1 = end1
+                start2 = end2
+        blockcount = len(blockSizes)
+        blockSizes = ",".join(map(str, blockSizes)) + ","
+        qStarts = ",".join(map(str, qStarts)) + ","
+        tStarts = ",".join(map(str, tStarts)) + ","
+        words = [str(match),
+                 str(mismatch),
+                 str(repmatch),
+                 str(Ns),
+                 str(Qgapcount),
+                 str(Qgapbases),
+                 str(Tgapcount),
+                 str(Tgapbases),
+                 strand,
+                 Qname,
+                 str(Qsize),
+                 str(Qstart),
+                 str(Qend),
+                 Tname,
+                 str(Tsize),
+                 str(Tstart),
+                 str(Tend),
+                 str(blockcount),
+                 blockSizes,
+                 qStarts,
+                 tStarts,
+                 ]
+        line = "\t".join(words) + "\n"
+        return line
+
+
+class Alignments(object):
```

Comment:
I'm not sure that this class name and the one-line docstring summary really convey the object's intended role. Does ``PairwiseAlignments`` seem any clearer?

Reply:
OK